### PR TITLE
Backup files to s3 Security-633

### PR DESCRIPTION
**Pull request by avinashneelakantam**
### Jira Story

- [Audit_logs_script_Security-529](https://osi-cwds.atlassian.net/browse/SECURITY-633)

## Description

DB Backups are moved to S3 Bucket and its is encrypted and secure. The backup job is monitored to ensure it runs per schedule and alerts are sent to DBA's and Service desk when any failure occurs. Access to S3 bucket restricted to DBAs
## Tests
- [ ] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [x] I have NOT included tests 

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] I have ran all tests for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: logging/src/scripts/Ldavis_access_script_security_628.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+ CREATE USER ldavis WITH
```
Comment (david-sanabria) [inline logging/src/scripts/Ldavis_access_script_security_628.sql]: This file should not be part of this PR.

**Comment by david-sanabria** [comment]
@mlipardo There are files in this PR that duplicate other PRs. I'm not clear why they are there. There is no description of the PR. I appreciate that you are using the standard github PR template; let's actually fill it out for reals.

File: logging/src/scripts/Ldavis_access_script_security_628.sql (hunk 1 above)
Comment (mlipardo) [reply logging/src/scripts/Ldavis_access_script_security_628.sql]: Yes, I'll do a rebase and start from the bottom PR then try remove some duplicate change files. I'll clean this up a bit. Thanks!

Comment (mlipardo) [reply logging/src/scripts/Ldavis_access_script_security_628.sql]: Done the clean up and only one file is merge. Thanks!

File: logging/src/scripts/db_backup_script.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+export host_name=$2
+export port_number=$3
+export user_name=$4
+export BUCKET=$5
+export INSTANCE=$6
+
+d=`date +%y%m%d%H%M%S`
+
+if (( "$#" < 6)); then
+    exit 2
```
Comment (david-sanabria) [inline logging/src/scripts/db_backup_script.sh]: Need an output  (e.g. `echo Missing arguments." >>&2` to explain why you exited since you use the same exit code throughout.

File: logging/src/scripts/db_backup_script.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+fi
+cd /opt/pgsql
+ PGPASSFILE=/opt/pgsql/.pgpass pg_dump -h $host_name -p $port_number -U $user_name $database_name > pg_dump_$INSTANCE$d.sql
```
Comment (david-sanabria) [inline logging/src/scripts/db_backup_script.sh]: What are the file permissions on the the `.pgpass` file? How was it created? Who has access to it?

File: logging/src/scripts/db_backup_script.sh [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+gzip pg_dump_$INSTANCE$d.sql
+ 
+ for i in `ls -lrt | awk '{print $9}' | grep pg_dump_` ; do
+   FILE_RUNDECK=`basename ${i}`
+aws s3 ls s3://${BUCKET}/${INSTANCE} --recursive | awk '{print $4}' | grep -o "pg_dump_*" > file_names_s3.out
+ file_name_s3=`cat file_names_s3.out | grep ${FILE_RUNDECK}`
+
+  if [ -z ${file_name_s3} ];
+then
+   echo "Downloading to S3 bucket ...."
```
Comment (david-sanabria) [inline logging/src/scripts/db_backup_script.sh]: Technically, you're *uploading* the file.

Comment (david-sanabria) [reply logging/src/scripts/db_backup_script.sh]: Please edit and add the name of the file being uploaded and the name of the S3 bucket.

File: logging/src/scripts/db_backup_script.sh (hunk 3 above)
Comment (avinashneelakantam) [reply logging/src/scripts/db_backup_script.sh]: Permissions to .pgpass are 0600

Comment (avinashneelakantam) [reply logging/src/scripts/db_backup_script.sh]: Rundeck user and Root has access to the file

File: logging/src/scripts/db_backup_script.sh [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+else 
+  echo "Backup succesfull"
+fi
+
+gzip pg_dump_$INSTANCE$d.sql
+ 
+for i in `ls -lrt | awk '{print $9}' | grep pg_dump_` ; do
+  FILE_RUNDECK=`basename ${i}`
+  aws s3 ls s3://${BUCKET}/${INSTANCE} --recursive | awk '{print $4}' | grep -o "pg_dump_*" > file_names_s3.out
+  file_name_s3=`cat file_names_s3.out | grep ${FILE_RUNDECK}`
```
Comment (david-sanabria) [inline logging/src/scripts/db_backup_script.sh]: add `$? || echo 'Download of file names failed'`

Comment (david-sanabria) [reply logging/src/scripts/db_backup_script.sh]: add `if [ ! "$?" -eq "0"]; then exit 3` so we trap any failures with the `aws` command
